### PR TITLE
feat: account icon improvements

### PR DESCRIPTION
**Pull request by fairlighteth**
# Summary

- Addresses https://github.com/cowprotocol/cowswap/issues/3696
- Makes sure to fallback to connection type icon (e.g. Walletconnect)
- Also solved another issue, where if you initially connected with a wallet it renders the Jazzicon (identicon). But after some time it will try to fetch the ENS name (if available) + any IPFS Avatar linked to the ENS. If that IPFS image doesn't resolve correctly you'd also see a blank image. Right now logic has been implemented to make sure the Jazzicon is always the fallback.

## Before
Screenshot 2024-01-30 at 11 44 26

Screenshot 2024-01-30 at 11 44 19

## After

Screenshot 2024-01-30 at 11 53 15

Screenshot 2024-01-30 at 11 18 12

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @fairlighteth , great!
But there is one more place where I can see a broken logo 
image
Could you please fix this?

File: apps/cowswap-frontend/src/modules/account/containers/AccountDetails/index.tsx [hunk 1]
```diff
@@ -80,9 +81,30 @@ export function renderActivities(activities: ActivityDescriptors[]) {
   )
 }
 
-export function getStatusIcon(connector: Connector, walletDetails?: WalletDetails, size?: number) {
+export function getStatusIcon(
```
Comment (shoom3301) [inline apps/cowswap-frontend/src/modules/account/containers/AccountDetails/index.tsx:84]: Would be better to convert the function to a react component and have image loading state inside

Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/account/containers/AccountDetails/index.tsx:84]: And extract it to a separate file of course

**Review by alfetopito — APPROVED** [review]
Worked great for me!
Just the comments from Sasha pending.

**Review by elena-zh — APPROVED** [review]
Hey @fairlighteth , all is great now. 
Moreover, this PR fixes #3431 issue . 
Thanks!

**Review by shoom3301 — APPROVED** [review]

File: apps/cowswap-frontend/src/modules/account/containers/AccountDetails/index.tsx (hunk 1 above)
Comment (shoom3301) [reply apps/cowswap-frontend/src/modules/account/containers/AccountDetails/index.tsx]: It looks perfect now, thanks!

**Pull request merged**